File: Baris dan Deret Aritmetika/Baris dan Deret Aritmetika/BarisdanDeretAritmetika.java
import java.util.Scanner;

public class BarisdanDeretAritmetika
{
    static Scanner scan = new Scanner (System.in);
    public static void main (String [] args)
    {
        int a = 0, b = 0, u1 = 0, u2 = 0, u3 = 0;
        byte pilihan = 0;

        do
        {
            System.out.println ("========== Baris dan Deret Aritmetika ==========");
            System.out.println ("1. Baris");
            System.out.println ("2. Deret");
            System.out.println ("3. Keluar");
            System.out.print ("Pilih salah satu: ");
            pilihan = scan.nextByte ();

            switch (pilihan)
            {
                case 1: baris (a, b, u1, u2, u3);    
                        break;
                case 2: deret ();
                        break;
                case 3: System.out.println ("Terima kasih telah menggunakan program ini!");
                        break;
                default: System.out.println ("Input error");
                         break;
            }
        } while (pilihan != 3);
    }
    public static void baris (int a, int b, int u1, int u2, int u3)
    {
        byte pilihan = 0;

        do
        {
            System.out.println ("==============================");
            System.out.println ("1. Suku ke-n");
            System.out.println ("2. Beda (selisih)");
            System.out.println ("3. Suku tengah");
            System.out.println ("4. Kembali");
            System.out.println ("5. Keluar");
            System.out.println ("==============================");
            System.out.print ("Pilih salah satu: ");
            pilihan = scan.nextByte ();

            switch (pilihan)
            {
                case 1: suku_ke_n (a, b);
                        break;
                case 2: beda (u1, u2, u3, b);
                        break;
                case 3: suku_tengah (a);
                        break;
                case 4: break;
                case 5: System.out.println ("Terima kasih telah menggunakan program ini!");
                        break;
                default: System.out.println ("Input error");
                         break;
            }
        } while (pilihan != 5);
    }
    public static void suku_ke_n (int a, int b)
    {
        // a + (n - 1) x b
        int n = 0;

        System.out.print ("Masukkan suku pertama: ");
        a = scan.nextInt ();
        System.out.print ("Masukkan suku yang dicari: ");
        n = scan.nextInt ();
        System.out.print ("Masukkan beda (selisih): ");
        b = scan.nextInt ();

        System.out.println("Suku ke " + n + " = " + a + (n - 1) * b);
    }
    public static void beda (int u1, int u2, int u3, int b)
    {
        byte pilihan = 0;

        do
        {
            System.out.println ("===== Mencari Beda (Selisih) =====");
            System.out.println ("1. Beda (selisih) statis");
            System.out.println ("2. Beda (selisih) bertahap");
            System.out.println ("3. Kembali");
            System.out.println ("4. Keluar");
            System.out.print ("Pilih salah satu: ");
            pilihan = scan.nextByte ();

            switch (pilihan)
            {
                case 1:
                    // b = U2 - U1
                    System.out.print ("Masukkan nilai suku pertama: ");
                    u1 = scan.nextInt ();
                    System.out.print ("Masukkan nilai suku kedua: ");
                    u2 = scan.nextInt ();
                    System.out.println ("Beda (selisih) = " + (u2 - u1));
                    break;
                case 2:
                    // b = (U3 - U2) - (U2 - U1)
                    System.out.print ("Masukkan nilai suku pertama: "); 
                    u1 = scan.nextInt ();
                    System.out.print ("Masukkan nilai suku kedua: "); 
                    u2 = scan.nextInt ();
                    System.out.print ("Masukkan nilai suku ketiga: ");    
                    u3 = scan.nextInt ();
                    System.out.println ("Nilai suku pertama                                : " + u1);
                    System.out.println ("Nilai suku kedua                                  : " + u2);
                    System.out.println ("Nilai suku ketiga                                 : " + u3);
                    System.out.println ("Beda (selisih) suku pertama & suku kedua          : " + (u2 - u1));
                    System.out.println ("Beda (selisih) suku kedua & suku ketiga           : " + (u3 - u2));
                    System.out.println ("Hasil                                             : " + b);
                case 3: break;
                case 4:
                    System.out.println ("Terima kasih telah menggunakan program ini!");
                    break;
                default:
                    System.out.println ("Input error");
                    break;
            }
        } while (pilihan != 4);
    }
    public static void suku_tengah (int a)
    {
        byte pilihan = 0;

        do
        {
            System.out.println ("===== Mencari Suku Tengah =====");
            System.out.println ("1. Nilai suku tengah");
            System.out.println ("2. Nilai urutan suku tengah");
            System.out.println ("3. Kembali");
            System.out.println ("4. Keluar");
            System.out.print ("Pilih salah satu: ");
            pilihan = scan.nextByte ();

            int un = 0, n = 0;

            switch (pilihan)
            {
                case 1:
                    // suku tengah = (suku pertama - suku terakhir) / 2
                    System.out.print ("Masukkan nilai suku pertama: ");
                    a = scan.nextInt ();
                    System.out.print ("Masukkan nilai suku terakhir: ");
                    un = scan.nextInt ();
                    System.out.println ("Nilai suku tengah: " + (a + un) / 2);
                    break;

                case 2:
                    // urutan suku tengah = (banyaknya suku + 1) / 2
                    System.out.print ("Masukkan banyaknya suku: ");
                    n = scan.nextInt ();
                    System.out.println ("Nilai urutan suku tengah: " + (n + 1) / 2);
                    break;
                case 3: break;
                case 4:
                    System.out.println ("Terima kasih telah menggunakan program ini!");
                    break;
                default:
                    System.out.println ("Input error");
                    break;
            }
        } while (pilihan != 4);
    }
    public static void deret ()
    {
        // n / 2 x (2 x a + (n - 1) x b)
        int n = 0, a = 0, b = 0;

        System.out.print ("Masukkan banyak suku: ");
        n = scan.nextInt ();
        System.out.print ("Masukkan nilai suku pertama: ");
        a = scan.nextInt ();
        System.out.print ("Masukkan beda (selisih): ");
        b = scan.nextInt ();

        System.out.println ("Jumlah suku pertama = " + n / 2 * (2 * a + (n - 1) * b));
    }
}

File: Baris dan Deret Aritmetika/Baris dan Deret Aritmetika/Baris-dan-Deret-Aritmetika.py
def main ():
    a, b, u1, u2, u3 = 0, 0, 0, 0, 0
    while True:
        print ("========== Baris dan Deret Aritmetika ==========")
        print ("1. Baris")
        print ("2. Deret")
        print ("3. Keluar")
        pilihan = (int (input("Pilih salah satu: ")))
        if pilihan == 1:
            baris (a, b, u1, u2, u3)
        elif pilihan == 2:
            deret ()
        elif pilihan == 3:
            print ("Terima kasih telah menggunakan program ini!")
        else:
            print ("Input error")

def baris (a, b, u1, u2, u3):
    while True:
        print ("==============================")
        print ("1. Suku ke-n")
        print ("2. Beda (selisih)")
        print ("3. Suku tengah")
        print ("4. Kembali")
        print ("5. Keluar")
        print ("==============================")
        pilihan = int (input ("Pilih salah satu: "))
        if pilihan == 1:
            suku_ke_n (a, b)
        elif pilihan == 2:
            beda (u1, u2, u3, b)
        elif pilihan == 3:
            suku_tengah (a)
        elif pilihan == 4:
            main ()
        elif pilihan == 5:
            print ("Terima kasih telah menggunakan program ini")
        else:
            print ("Input error")

def suku_ke_n (a, b):
    # a + (n - 1) x b
    a = int (input ("Masukkan suku pertama: "))
    n = int (input ("Masukkan suku yang dicari: "))
    b = int (input ("Masukkan beda (selisih): "))
    print (f"Suku ke {n} = {a + (n - 1) * b}")

def beda ():
    while True:
        print ("===== Mencari Beda (Selisih) =====")
        print ("1. Beda (selisih) statis")
        print ("2. Beda (selisih) bertahap")
        print ("3. Kembali")
        print ("4. Keluar")
        print ("Pilih salah satu: ")
        pilihan = int (input ("Pilih salah satu: "))
        if pilihan == 1:
            # b = U2 - U1
            u1 = int (input ("Masukkan nilai suku pertama: "))
            u2 = int (input ("Masukkan nilai suku kedua: "))
            print (f"Beda (selisih) = {u2 - u1}")
        elif pilihan == 2:
            # b = (U3 - U2) - (U2 - U1)
            u1 = int (input ("Masukkan nilai suku pertama: "))
            u2 = int (input ("Masukkan nilai suku kedua: "))
            u3 = int (input ("Masukkan nilai suku ketiga: "))
            print (f"Nilai suku pertama                                : {u1}")
            print (f"Nilai suku kedua                                  : {u2}")
            print (f"Nilai suku ketiga                                 : {u3}")
            print (f"Beda (selisih) suku pertama & suku kedua          : {u2 - u1}")
            print (f"Beda (selisih) suku kedua & suku ketiga           : {u3 - u2}")
            b = (u3 - u2) - (u2 - u1)
            print (f"Hasil                                             : {b}")
        elif pilihan == 3:
            main ()
        elif pilihan == 4:
            print ("Terima kasih telah menggunakan program ini!")
        else:
            print ("Input error")

def suku_tengah (a):
    while True:
        print ("===== Mencari Suku Tengah =====")
        print ("1. Nilai suku tengah")
        print ("2. Nilai urutan suku tengah")
        print ("3. Kembali")
        print ("4. Keluar")
        pilihan = int (input ("Pilih salah satu: "))
        if pilihan == 1:
            # suku tengah = (suku pertama - suku terakhir) / 2
            a = int (input ("Masukkan nilai suku pertama: "))
            un = int (input ("Masukkan nilai suku terakhir: "))
            print (f"Nilai suku tengah: {(a + un) / 2}")
        elif pilihan == 2:
            # urutan suku tengah = (banyaknya suku + 1) / 2
            n = int (input ("Masukkan banyaknya suku: "))
            print (f"Nilai urutan suku tengah: {(n + 1) / 2}")
        elif pilihan == 3:
            main ()
        elif pilihan == 4:
            print ("Terima kasih telah menggunakan program ini!")
        else:
            print ("Input error")

def deret ():
    # n / 2 x (2 x a + (n - 1) x b)
    n = int (input ("Masukkan banyak suku: "))
    a = int (input ("Masukkan nilai suku pertama: "))
    b = int (input ("Masukkan beda (selisih): "))
    print (f"Jumlah suku pertama = {n / 2 * (2 * a + (n - 1) * b)}")

if __name__ == "__main__":
    main ()